apps/evm/src/store/index.ts

```diff
@@ -36,25 +37,22 @@ const useStore = create<State>()(
     set => ({
       userSettings: initialUserSettings,
       setUserSettings: ({ settings, chainIds = allChainIds }) =>
-        set(state => {
-          const newUserSettings = chainIds.reduce<Partial<UserChainSettings>>(
-            (acc, chainId) => ({
-              ...acc,
-              [chainId]: {
-                ...settings,
-              },
-            }),
-            {},
-          );
-
-          return {
-            ...state,
-            userSettings: {
-              ...state.userSettings,
-              ...newUserSettings,
-            },
-          };
-        }),
+        set(state => ({
+          ...state,
+          userSettings: {
+            ...state.userSettings,
+            ...chainIds.reduce<Partial<UserChainSettings>>(
+              (acc, chainId) => ({
+                ...acc,
+                [chainId]: {
+                  ...state.userSettings[chainId],
+                  ...settings,
+                },
+              }),
+              {},
+            ),
+          },
+        })),
```

Comment:
nit: maybe we should consider something like https://zustand.docs.pmnd.rs/integrations/immer-middleware

Reply:
I didn't know about Immer, that's nice. I've updated the logic to use it, it's indeed cleaner.